lib/rspec/core/example.rb

```diff
@@ -227,14 +227,14 @@ def run(example_group_instance, reporter)
               rescue Pending::SkipDeclaredInExample
                 # no-op, required metadata has already been set by the `skip`
                 # method.
-              rescue Exception => e
+              rescue AllExceptionsExcludingDangerousOnesOnRubiesThatAllowIt => e
                 set_exception(e)
               ensure
                 run_after_example
               end
             end
           end
-        rescue Exception => e
+        rescue Support::AllExceptionsExceptOnesWeMustNotRescue => e
```

Comment:
The one above is `RSpec::Core::AllExceptionsExcludingDangerousOnesOnRubiesThatAllowIt`, not a constant from the `RSpec::Support` namespace.  See the definition and comment below.  Yes, it's weird.